lib/OpenQA/Scheduler/Scheduler.pm

```diff
@@ -224,8 +224,12 @@ sub schedule {
 
                 # NOTE: $worker->connected is too much expensive since is over dbus, prefer dead.
                 # shuffle avoids starvation if a free worker keeps failing.
-                my @free_workers
-                  = shuffle(grep { !$_->dead } schema->resultset("Workers")->search({job_id => undef})->all());
+                my @free_workers = shuffle(
+                    grep {
+                            !$_->dead
+                          && $_->get_websocket_api_version()
+                          == OpenQA::WebSockets::Server->INTERFACE_VERSION
```

Comment:
This will make sure that all the workers that are free for job allocating will have the same websocket version